pkg/reconciler/image/reconcile_build.go

```diff
@@ -18,16 +18,22 @@ func (c *Reconciler) reconcileBuild(image *v1alpha1.Image, latestBuild *v1alpha1
 		return v1alpha1.ImageStatus{}, err
 	}
 
-	reasons, needed := buildNeeded(image, latestBuild, sourceResolver, builder)
-	switch needed {
+	buildDeterminer := NewBuildDeterminer(image, latestBuild, sourceResolver, builder)
```

Comment:
In golang world the construction of a `BuildDeterminer` object seems a bit unnecessary. Could we just call a method called determineBuild which returns a build result that knows if a build is needed, etc?

Reply:
marking as resolved. addressed in https://github.com/pivotal/kpack/commit/e6e1245a58ff31c160a2607f6f7d9aac1092993f